test/common/stats/thread_local_store_test.cc

```diff
@@ -90,6 +90,116 @@ class StatsThreadLocalStoreTest : public testing::Test, public RawStatDataAlloca
   std::unique_ptr<ThreadLocalStoreImpl> store_;
 };
 
+class HistogramTest : public testing::Test, public RawStatDataAllocator {
+public:
+  void SetUp() override {
+    ON_CALL(*this, alloc(_)).WillByDefault(Invoke([this](const std::string& name) -> RawStatData* {
+      return alloc_.alloc(name);
+    }));
+
+    ON_CALL(*this, free(_)).WillByDefault(Invoke([this](RawStatData& data) -> void {
+      return alloc_.free(data);
+    }));
+
+    EXPECT_CALL(*this, alloc("stats.overflow"));
+    store_.reset(new ThreadLocalStoreImpl(*this));
+    store_->addSink(sink_);
+    InSequence s;
```

Comment:
I admit I've never been a huge gmock user.  But I looked up InSequence and I suspect this usage model is not quite right.  Check out https://github.com/google/googletest/blob/master/googlemock/docs/ForDummies.md#ordered-vs-unordered-calls

I think this only affects what's in the scope of that declaration, which in this case is only one call.  I would think it should either be declared at the top of SetUp or as a class member var.